lib/util/probe.js

```diff
@@ -33,6 +34,14 @@ function startProbeServer(config, callback) {
     probeServer.start();
 }
 
+/**
+ * Start probe server for Queue Processor
+ * @param {ProbeServerConfig} config - Configuration for probe server
+ * @returns {Promise|ProbeServer} Probe server or undefined if disabled
+ */
+const startProbeServerPromise = util.promisify(startProbeServer);
```

Comment:
not sure introducing this method makes the code more readable...

A better way may be to update the `startProbeServer` function to return a promise when no callback is provided:

```
// @return {Promise|undefined}
function startProbeServer(config, callback) {
   if (callback == undefined)
     return util.promisify(startProbeServer)(config);

  // else return legacy code
  [...]
}